function-executor/src/main.rs

```diff
@@ -18,22 +17,19 @@ use anna::{
 use anyhow::Context;
 use essa_common::{
     essa_default_zenoh_prefix, executor_run_function_subscribe_topic, executor_run_module_topic,
-    scheduler_function_call_topic,
 };
 use std::{
-    collections::HashMap,
     sync::Arc,
     thread,
     time::{Duration, Instant},
 };
 use uuid::Uuid;
-use wasmtime::{Caller, Engine, Extern, Linker, Module, Store, Trap, ValType};
-use wasmtime_wasi::{WasiCtx, WasiCtxBuilder};
-use zenoh::{
-    prelude::{Receiver, Sample, SplitBuffer, ZFuture},
-    query::ReplyReceiver,
-    queryable::Query,
-};
+use zenoh::prelude::{Receiver, SplitBuffer, ZFuture};
+
+#[cfg(not(feature = "wasmtime_executor"))]
+mod wasmedge;
+#[cfg(feature = "wasmtime_executor")]
+mod wasmtime;
```

Comment:
I think it would be better to throw a compile error if multiple runtimes are enabled:

```suggestion
#[cfg(all(feature = "wasmedge_executor", feature = "wasmtime_executor"))]
compile_error!(
    "Both `wasmedge_executor` and `wasmtime_executor` features are enabled, but \
    only one WASM runtime is supported at a time"
);
#[cfg(all(not(feature = "wasmedge_executor"), not(feature = "wasmtime_executor")))]
compile_error!("Either the `wasmedge_executor` or the `wasmtime_executor` feature must be enabled");
#[cfg(feature = "wasmedge_executor")]
mod wasmedge;
#[cfg(feature = "wasmtime_executor")]
mod wasmtime;
```